src/Shared/UnitTests/FileMatcher_Tests.cs

```diff
@@ -995,8 +995,14 @@ private void GetMatchingDirectories(string[] candidates, string path, string pat
                             int nextSlash = normalizedCandidate.IndexOfAny(FileMatcher.directorySeparatorCharacters, path.Length + 1);
                             if (nextSlash != -1)
                             {
-                                string match = normalizedCandidate.Substring(0, nextSlash + 1);
+                                string match;
+
+                                match = normalizedCandidate.StartsWith(@"\\") && string.IsNullOrEmpty(path)
+                                    ? normalizedCandidate.Substring(0, nextSlash + 1)
+                                    : normalizedCandidate.Substring(0, nextSlash);
```

Comment:
No, but this code is trying to parse paths and the initial +1 enabled:
- correct parsing of paths starting with `\\`
- incorrect handling of paths, causing a `\` to remain there

By doing the if I kept the correct case and removed the incorrect one.